Microsoft.Toolkit.Uwp/Helpers/BackgroundTaskHelper.cs

```diff
@@ -0,0 +1,175 @@
+    /// </summary>
+    public static class BackgroundTaskHelper
+    {
+
+        /// <summary>
+        /// Check if a background task is registered.
+        /// </summary>
+        /// <param name="backgroundTaskName">The name of the background task class</param>
+        /// <returns>True/False indicating if a background task was registered or not</returns>
+        public static bool IsBackgroundTaskRegistered(string backgroundTaskName)
+        {
+            return BackgroundTaskRegistration.AllTasks.Any(t => t.Value.Name == backgroundTaskName);
+        }
+
+        /// <summary>
+        /// Check if a background task is registered.
+        /// </summary>
+        /// <param name="backgroundTaskType">The type of the background task. This class has to implement IBackgroundTask</param>
+        /// <returns>True/False indicating if a background task was registered or not</returns>
+        public static bool IsBackgroundTaskRegistered(Type backgroundTaskType)
+        {
+            return IsBackgroundTaskRegistered(backgroundTaskType.Name);
+        }
+
+        /// <summary>
+        /// Register a background task with conditions.
+        /// If the task is already registered and has the same trigger, returns the existing registration if it has the same trigger.
+        /// If the task is already registered but has different trigger, return null by default.
+        /// Or set <paramref name="forceRegister"/> to true to un-register the old one and then re-register.
+        /// </summary>
+        /// <param name="backgroundTaskName">Name of the background task class</param>
+        /// <param name="backgroundTaskEntryPoint">Entry point of the background task. This parameter will not be used if <paramref name="isSingleProcess"/> is true</param>
+        /// <param name="trigger">Trigger that indicate when the background task should be invoked</param>
+        /// <param name="forceRegister">Indicate if the background task will be force installed in the case of being already registered</param>
+        /// <param name="isSingleProcess">Indicate if we are registering under Single Process Model. <paramref name="backgroundTaskEntryPoint"/> won't be used if this is true. Default is false.
+        /// WARNING: Single Process Model only works with Windows 10 Anniversary Update (14393) and later.</param>
+        /// <param name="enforceConditions">Indicate if the background task should quit if condition is no longer valid</param>
+        /// <param name="conditions">Optional conditions for the background task to run with</param>
+        /// <returns>Background Task that was registered with the system</returns>
+        public static BackgroundTaskRegistration Register(string backgroundTaskName, string backgroundTaskEntryPoint, IBackgroundTrigger trigger, bool forceRegister = false, bool isSingleProcess = false, bool enforceConditions = true, params IBackgroundCondition[] conditions)
```

Comment:
I hate to request yet another change, but ultimately we need a simple method for Single Process Model - this method has way too many parameters. "isSingleProcess" is redundant, since it can be implied by whether a backgroundTaskEntryPoint was provided, for example.

For the public API, I think we need something cleaner.

Here's what I propose - get rid of the "backgroundTaskEntryPoint" and "isSingleProcess", and have a method that's dedicated to SPM

    public static BackgroundTaskRegistration Register(
        string backgroundTaskName, 
        IBackgroundTrigger trigger, 
        bool forceRegister = false,
        bool enforceConditions = true, 
        params IBackgroundCondition[] conditions)

This will be the SPM method devs can use. Hopefully it will become the go-to method. In the summary text, we'll explain that "This registers under the Single Process Model. WARNING: Single Process Model only works with Windows 10 Anniversary Update (14393) and later."

Then we have the MPM method, using the Type (notice that we remove "isSingleProcess"). Similarly, the summary text will explain that this uses the normal non-Single Process Model background tasks.

    public static BackgroundTaskRegistration Register( 
        Type backgroundTaskType, 
        IBackgroundTrigger trigger, 
        bool forceRegister = false, 
        bool enforceConditions = true,  
        params IBackgroundCondition[] conditions)

And finally, the shared method among the two, which we could keep internal, or choose to expose...

    public static BackgroundTaskRegistration Register(
        string backgroundTaskName, 
        IBackgroundTrigger trigger, 
        bool forceRegister = false,
        bool enforceConditions = true, 
        params IBackgroundCondition[] conditions)

The benefit of exposing it would be if devs want to use MPM but don't want to use the Type option, they can specify their own name and entry point. But the cost is that there's now three overloads, which adds confusion as to which you should pick. So I'm indifferent on that, I could go either way.

Thoughts on all of that? @cc-cortes, @Librazy, @deltakosh, @ScottIsAFool? Does that sound cleaner?